lib/utils.js

```diff
@@ -35,4 +35,42 @@ function isStream (stream) {
   return stream !== null && typeof stream === 'object' && typeof stream.pipe === 'function'
 }
 
-module.exports = { isGzip, isDeflate, isStream }
+// It is a helper used to provide a async iteratable for
+// Readable.from
+async function * intoAsyncIterator (payload) {
+  const isBuffer = Buffer.isBuffer(payload)
+
+  if (
+    (
+    // ArrayBuffer
+      payload instanceof ArrayBuffer ||
+    // NodeJS.TypedArray
+    ArrayBuffer.isView(payload)
+    ) &&
+    // Exclude Buffer to prevent double cast
+    !isBuffer
+  ) {
+    payload = Buffer.from(payload)
+  }
+
+  // Iterator
+  if (typeof payload === 'object' && typeof payload[Symbol.iterator] === 'function' && !isBuffer) {
+    for (const chunk of payload) {
+      yield chunk
+    }
+    return
+  }
+
+  // Async Iterator
+  if (typeof payload === 'object' && typeof payload[Symbol.asyncIterator] === 'function' && !isBuffer) {
```

Comment:
Can asyncIterator be something else than a function?

Reply:
No, the symbol should always be function. So, it can be loop.

Reply:
Can we do something like Symbol.asyncIterator in payload? Should be faster than typeof.